docs/enhancements/0001-worker-pods.md

```diff
@@ -0,0 +1,229 @@
+                                       │ Nodes ◄──┐
+                                       │       │  │
+                                       └───────┘  │ watches
+                                                  │
+                                     ┌─────────┐  │
+                                     │         │  │
+                                     │ Modules │  │     ┌───────────────────────┐
+                                     │         │  │     │                       │
+                                     └─────▲───┘  │     │ ┌───────────────────┐ │
+                                           │      │     │ │                   │ │    creates & owns
+                                watches &  └──────┴─────┼─┤ Module controller ├─┼─────────────────────┐
+                                updates status          │ │                   │ │                     │
+                                                        │ └───────────────────┘ │           ┌─────────▼────────┐
+                                                        │                       │           │                  │
+                                                        │                       │           │ NodeModulesState │
+                                                        │ ┌──────────────────┐  │           │                  │
+              ┌───────────┐                             │ │                  │  │           └─────────▲────────┘
+              │           │        creates & owns;      │ │ NodeModulesState │  │                     │
+              │ ConfigMap ◄─────────────────────────────┼─┤ controller       ├──┼─────────────────────┘
+              │           │   writes worker config &    │ │                  │  │   watches &
+              └─────▲─────┘   reads worker output from  │ └──┬───────────────┘  │   updates status
+                    │                                   │    │                  │
+            mounts; │                                   │    │     KMM operator │
+reads config from & │       ┌────────────┐              └────┼──────────────────┘
+   writes output to │       │            │                   │
+                    └───────┤ KMM worker ◄───────────────────┘
+                            │    Pod     │ creates & owns
+                            │            │
+                            └──────┬─────┘
+                                   │
+                                   │
+                                   │               ┌──────────────────────┐
+                                   │               │                      │
+                                   └───────────────► registry.example.org │
+                                      pulls from   │                      │
+                                                   └──────────────────────┘
+```
+
+#### The operator
+
+The operator runs two main controllers: the `Module` controller and the `NodeModulesState` controller.
```

Comment:
Why do we need this new controller? Why don't we go directly from `Module` to worker pods? I can understand the separation of concerns but isn't it just exposing to the cluster another object which is just an intermediate one?

Reply:
NodeModulesState is basically a current state and config of the Module on the Node. We don''t want to store all that info in the Module status, since it is a lot of info that user does not need, and it will also means that any "bug fix"/development feature" will result in user API change.